include/fmt/format.h

```diff
@@ -566,7 +566,7 @@ template <typename U>
 void buffer<T>::append(const U* begin, const U* end) {
   size_t new_size = size_ + to_unsigned(end - begin);
   reserve(new_size);
-  std::uninitialized_copy(begin, end, make_checked(ptr_, capacity_) + size_);
+  std::uninitialized_copy(begin, end, make_checked(ptr_ + size_, capacity_));
```

Comment:
Thanks for the PR. I think you also need to subtract `size_` from `capacity_`.

Reply:
Will do, but why do all the test pass? :)

Reply:
Because we don't write over capacity. The whole checked iterator machinery in MSVC is completely useless and is here mostly to avoid warnings but let's provide correct bounds anyway.